frontend/src/metabase/common/hooks/use-entity-list-query/use-entity-list-query.unit.spec.tsx

```diff
@@ -0,0 +1,168 @@
+      getLoaded: Databases.selectors.getLoaded,
+      getError: Databases.selectors.getError,
+    },
+  );
+
+  const dispatch = useDispatch();
+  const handleInvalidate = () => dispatch(Databases.actions.invalidateLists());
+
+  if (isLoading || error) {
+    return <LoadingAndErrorWrapper loading={isLoading} error={error} />;
+  }
+
+  return (
+    <div>
+      <button onClick={handleInvalidate}>Invalidate databases</button>
+      {data.map(database => (
+        <div key={database.id}>{database.name}</div>
+      ))}
+      <TestInnerComponent />
+    </div>
+  );
+};
+
+const TestInnerComponent = () => {
+  const {
+    data = [],
+    isLoading,
+    error,
+  } = useEntityListQuery<Table, TableListQuery>(
+    {
+      reload: true,
+    },
+    {
+      fetchList: Tables.actions.fetchList,
+      getList: Tables.selectors.getList,
+      getLoading: Tables.selectors.getLoading,
+      getLoaded: Tables.selectors.getLoaded,
+      getError: Tables.selectors.getError,
+    },
+  );
+
+  const dispatch = useDispatch();
+  const handleInvalidate = () => dispatch(Tables.actions.invalidateLists());
+
+  if (isLoading || error) {
+    return <LoadingAndErrorWrapper loading={isLoading} error={error} />;
+  }
+
+  return (
+    <div>
+      <button onClick={handleInvalidate}>Invalidate tables</button>
+      {data.map(table => (
+        <div key={table.id}>{table.name}</div>
+      ))}
+    </div>
+  );
+};
+
+const setup = () => {
+  setupDatabasesEndpoints([TEST_DB]);
+  setupTablesEndpoints([TEST_TABLE]);
+  return renderWithProviders(<TestComponent />);
+};
+
+describe("useEntityListQuery", () => {
+  it("should be initially loading", () => {
+    setup();
+
+    expect(screen.getByText("Loading...")).toBeInTheDocument();
+  });
+
+  it("should initially load data only once the reload flag in a nested component tree", async () => {
+    setup();
+
+    await waitForElementToBeRemoved(() => screen.queryByText("Loading..."));
+
+    expect(screen.getByText(TEST_DB.name)).toBeInTheDocument();
+    expect(screen.getByText(TEST_TABLE.name)).toBeInTheDocument();
+    expect(fetchMock.calls("path:/api/database")).toHaveLength(1);
+    expect(fetchMock.calls("path:/api/table")).toHaveLength(1);
+  });
+
+  it("should not reload data when re-rendered", async () => {
+    const { rerender } = setup();
+
+    await waitForElementToBeRemoved(() => screen.queryByText("Loading..."));
+    rerender(<TestComponent />);
+
+    expect(screen.getByText(TEST_DB.name)).toBeInTheDocument();
+    expect(screen.getByText(TEST_TABLE.name)).toBeInTheDocument();
+    expect(fetchMock.calls("path:/api/database")).toHaveLength(1);
+    expect(fetchMock.calls("path:/api/table")).toHaveLength(1);
+  });
+
+  it("should reload data only for calls with the reload flag when re-mounted", async () => {
+    const { rerender } = setup();
+
+    await waitForElementToBeRemoved(() => screen.queryByText("Loading..."));
+    rerender(<div />);
```

Comment:
nit: we need to make sure `Loading...` appeared between the second re-render

Reply:
Yes there is a check for this already because `waitForElementToBeRemoved` throws if the element wasn't initially there